### PR TITLE
MAYA-109380 - Create a common parent class for HdVP2BasisCurves, HdVP2Mesh and future VP2 shapes

### PR DESCRIPTION
**Pull request by vlasovi**
Part one of code refactoring to get rid of code duplication in HdVP2BasisCurves and HdVP2Mesh

### CODE REVIEW

File: lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp [hunk 1]
```diff
@@ -55,77 +55,6 @@ const TfTokenVector sFallbackShaderPrimvars
     = { HdTokens->displayColor, HdTokens->displayOpacity, HdTokens->normals };
 
 const MColor       kOpaqueBlue(0.0f, 0.0f, 1.0f, 1.0f); //!< Opaque blue
```
Comment (williamkrick) [inline lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp:57]: MayaUsdRPrim has a kOpaqueBlue member.

**Review by williamkrick — APPROVED** [review]
Looks great! I love seeing all that code get removed.

File: lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp [hunk 2]
```diff
@@ -54,78 +54,7 @@ namespace {
...
-    //! returns true if there is no state to commit
-    bool Empty()
-    {
-        return _indexBufferData == nullptr && _shader == nullptr && _enabled == nullptr
-            && !_geometryDirty && _boundingBox == nullptr && !_renderItemData._usingInstancedDraw
-            && _instanceTransforms.length() == 0 && _ufeIdentifiers.length() == 0
-            && _worldMatrix == nullptr;
-    }
-};
+const MColor kOpaqueBlue(0.0f, 0.0f, 1.0f, 1.0f); //!< Opaque blue
```
Comment (williamkrick) [inline lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp:57]: There is already MayaUsdRPrim::kOpaqueBlue so we don't need another version of it here.

https://github.com/Autodesk/maya-usd/pull/2076/files#diff-3be343169a56ae82491707b0790e28079ca0357ce91914123ec0cdd222e9c0c2R156

**Review by williamkrick — APPROVED** [review]

File: lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp (hunk 2 above)
Comment (vlasovi) [reply lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp:57]: I'll take care of that in the following PR

**Comment by vlasovi** [comment]
Closing the PR as the feature will continue to be developed in a separate branch: vlasovi/points

**Pull request closed without merging**